solr/CHANGES.txt

```diff
@@ -186,6 +186,9 @@ Other Changes
 
 * SOLR-14034: Remove deprecated min_rf references (Tim Dillon)
 
+* SOLR-14067: StatelessScriptUpdateProcessor moved to it's own /contrib/scripting/ package instead
+ of shipping as part of Solr due to security concerns.  Renamed to ScriptingUpdateProcessor. (Eric Pugh)
```

Comment:
```suggestion
 of shipping as part of Solr due to security concerns.  Renamed to ScriptUpdateProcessorFactory. (Eric Pugh)
```

Reply:
I don't love that we have the `Factory` suffix, as that feels like an implementation detail of how update processors work.

I wish we could refer to this as the `ScriptUpdateProcessor`, even though this specific class is buried in the file `ScriptUpdateProcessorFactory.java` as:

```
private static class ScriptUpdateProcessor extends UpdateRequestProcessor
```

Would it be worth pulling the inner class to it's own `ScriptUpdateProcessor.java` file?   

Then we could just link to that file everywhere.  Thoughts????

Reply:
I agree 100% on seeing \*Factory all over the config being poor for the reason you gave.  It could also be argued that even the "UpdateProcessor" part is quite redundant based on where we declare it.  Have you noticed changes in Lucene to how schema analysis components are resolved, affecting the Solr schema (master only)?  See solr/server/solr/configsets/_default/conf/managed-schema -- `<filter name="lowercase"/>` it's beautiful.    No "FilterFactory" suffix.  Eventually I hope we can take the same approach throughout Solr.  Lucene uses an SPI approach which means a special file listing each implementation.  Something like that could be embraced.  No need to separate the factory from inner class over this.

Reply:
Yep.   I didn't know about that feature.   So, as far as this specific PR goes, can we leave it as ScriptUpdateProcessor, but of course the Javadoc links go to a ScriptUpdateProcessorFactory.html file?